src/main/kotlin/com/exactpro/th2/codec/xml/XmlPipelineCodec.kt

```diff
@@ -77,18 +98,33 @@ open class XmlPipelineCodec(private val settings: XmlPipelineCodecSettings, xsdM
         }.build()
     }
 
+    private fun encodeOne(message: ParsedMessage): RawMessage {
+        val map = message.body
+        val xmlString = Xml.toXml(map)
+
+        validator.validate(xmlString.toByteArray())
+        LOGGER.debug { "Validation of incoming parsed message complete: ${message.type}" }
+
+        return RawMessage(
+            id = message.id,
+            eventId = message.eventId,
+            metadata = message.metadata,
+            protocol = XmlPipelineCodecFactory.PROTOCOL,
+            body = Unpooled.copiedBuffer(xmlString, xmlCharset)
```

Comment:
Do we need to copy the result bytes? I think we can wrap the bytes that we used for `validator.validate` method call